### ISSUE TITLE
Add roles to members based on conditionals

### ISSUE THREAD

**Issue by tjrgg** [typicalbot/typicalbot#20]
## Issue

- [x] I have checked for related issues.
- [ ] I have updated to the latest [TypicalBot version](https://github.com/typicalbot/typicalbot).
- [ ] I am using an official bot operated by TypicalBot.

### Issue Type

- [ ] Bug Report
- [x] Feature Request

### Description

A subcommand for the `roles` command that adds roles to all members that meet a condition (has a role, for example).

**Comment by nsylke** [comment]
I like the idea, but not sure on the implementation. I'd also be interested in the usage/arguments of the command, I feel like this `roles give all Verified has:Subscriber` is quite bad.

**Comment by LittleKitacho** [comment]
Syntax might be like `roles give Verified Subscriber`, verified being the role to give and subscriber the role to get members and give the role.

**Comment by tjrgg** [comment]
> I like the idea, but not sure on the implementation. I'd also be interested in the usage/arguments of the command, I feel like this `roles give all Verified has:Subscriber` is quite bad.
I like where you're going with that. You could make the conditionals optional and that syntax could also easily support multiple conditionals.

> Syntax might be like `roles give Verified Subscriber`, verified being the role to give and subscriber the role to get members and give the role.
This supports only the condition of having a role. While that would probably good enough, I was looking to expand into other conditions with this suggestion. For instance, why not have a condition of not having a role?

**Comment by LittleKitacho** [comment]
Yeah.  Add some basic key word identifier, like HAS or NOT or AND or OR.

**Comment by nsylke** [comment]
> 
> I like where you're going with that. You could make the conditionals optional and that syntax could also easily support multiple conditionals.

Fair enough, I just don't know if people will get _used_ to this type of syntax. I feel like it might over-complicated the command. 

> 
> Yeah. Add some basic key word identifier, like HAS or NOT or AND or OR.

I like these. Probably would be the first iteration of conditions, if implemented.

**Comment by nsylke** [comment]
I should also note that I've been working on conditions in our upcoming command framework. I feel like adding the implementation into our framework will be a bit easier than trying to implement in our current command system.

![Commander](https://cdn.discordapp.com/attachments/533047058232377380/554383584186728476/unknown.png)

**Comment by tjrgg** [comment]
> I should also note that I've been working on conditions in our upcoming command framework. I feel like adding the implementation into our framework will be a bit easier than trying to implement in our current command system.
> 
> ![Commander](https://camo.githubusercontent.com/341916317e56aedc3478cf4c9ac9d1ae0339c22e/68747470733a2f2f63646e2e646973636f72646170702e636f6d2f6174746163686d656e74732f3533333034373035383233323337373338302f3535343338333538343138363732383437362f756e6b6e6f776e2e706e67)

Will those conditions be dynamic? Would they be configurable per guild?

**Comment by nsylke** [comment]
> 
> 
> > I should also note that I've been working on conditions in our upcoming command framework. I feel like adding the implementation into our framework will be a bit easier than trying to implement in our current command system.
> > ![Commander](https://camo.githubusercontent.com/341916317e56aedc3478cf4c9ac9d1ae0339c22e/68747470733a2f2f63646e2e646973636f72646170702e636f6d2f6174746163686d656e74732f3533333034373035383233323337373338302f3535343338333538343138363732383437362f756e6b6e6f776e2e706e67)
> 
> Will those conditions be dynamic? Would they be configurable per guild?

It would be possible, but not for all conditions. Like the image above, it wouldn't make sense to make the condition to be changed by Guild Settings.

![Parameter Condition](https://cdn.discordapp.com/attachments/189578968410750977/570682984978448405/unknown.png)

However, in this image you can see `@Default("new role")`, which could be easily changed to use Guild Settings over our predefined value. *It wouldn't really make sense to make this option configurable, but it's an example.*

**Comment by tjrgg** [comment]
It wouldn't make sense to make the `eval` command configurable via Guild Settings, but it could be configurable using some form of Global Settings.

**Comment by nsylke** [comment]
> 
> 
> It wouldn't make sense to make the `eval` command configurable via Guild Settings, but it could be configurable using some form of Global Settings.

Mhm, we current have configuration files. Yet, I feel it would be more ideal to store these in the database and add a command to change these settings. In fact, part of the options could be included in the setup process.

**Comment by tjrgg** [comment]
Allow to mass assign/unassign roles from users (suggested by UserID10T#2790, #51).
